### PR TITLE
Add wagtail-localize-smartling to the project, for Smartling L10N support

### LINKED ISSUE #14774: Integrate and evaluate wagtail-localize-smartling
## Description

v0.1 of wagtail-localize-smartling is available, so we should try it out and feed back requests

---

#### Success Criteria
- [x] app installed and working
- [x] feedback formulated
- [x] feedback delivered

### PR DESCRIPTION
**Pull request by stevejalim**
## One-line summary

This changeset drops in wagtail-localize-smartling, which will flow strings from Wagtail to Smartling for L10N and then back again. 

## Significant changes and points to review

0.2.3 is an early-ish, Minimum Lovable Product version, and there may well be some fast-follow changes from the vendor - those are in discussion at the mo, and we can pick up anything else we want, as it's "our" codebase now.

I've tested it manually, locally, for now, with sandbox-y API keys and it works for the core cases. There are some rough edges but we'll sort them out.

The infra PR needed to accompany these changes will need to:

1) Set the appropriate env vars [as documented](https://github.com/mozilla/wagtail-localize-smartling?tab=readme-ov-file#installation)
2) Set a cron job that runs ./manage.py sync_smartling every X minutes (15?). Doing it as a separate job on a cron means we don't risk timeouts in the web UI.

## Issue / Bugzilla link

Resolves #14774 

## Testing

N/A for now - I'll record a video demo, but waiting to see if there'll be a 0.2.4 soon (or not yet)

### CODE REVIEW

**Comment by stevejalim** [comment]
Review tagging is more about awareness than close reviewing right now. Good to get this merged (when the infra is ready)

**Review by linjinguang39 — APPROVED** [review]

File: bedrock/settings/base.py [hunk 1]
```diff
@@ -2118,6 +2119,24 @@ def lazy_wagtail_langs():
...
+    "PROJECT_ID": config("WAGTAIL_LOCALIZE_SMARTLING_PROJECT_ID", default="setme"),
+    "USER_IDENTIFIER": config("WAGTAIL_LOCALIZE_SMARTLING_USER_IDENTIFIER", default="setme"),
+    "USER_SECRET": config("WAGTAIL_LOCALIZE_SMARTLING_USER_SECRET", default="setme"),
+    # Optional settings and their default values
+    "REQUIRED": config(
+        "WAGTAIL_LOCALIZE_SMARTLING_ALWAYS_SEND",
+        default="False",
+        parser=bool,
+    ),  # Set this to True to always send translations to Smartling
+    "ENVIRONMENT": config("WAGTAIL_LOCALIZE_SMARTLING_ALWAYS_SEND", default="production"),  # Set this to "staging" to use Smartling's staging API
```
Comment (robhudson) [inline bedrock/settings/base.py:2134]: This one and the one above use the same env var?

**Comment by stevejalim** [comment]
Secrets deployed to dev and will not crash when code reaches Stage or Prod because we have non-exploding (but deliberately invalid) defaults set of `setme`

File: bedrock/settings/base.py [hunk 2]
```diff
@@ -2118,6 +2119,29 @@ def lazy_wagtail_langs():
...
+    "USER_IDENTIFIER": config("WAGTAIL_LOCALIZE_SMARTLING_USER_IDENTIFIER", default="setme"),
+    "USER_SECRET": config("WAGTAIL_LOCALIZE_SMARTLING_USER_SECRET", default="setme"),
+    # Optional settings and their default values
+    "REQUIRED": config(
+        "WAGTAIL_LOCALIZE_SMARTLING_ALWAYS_SEND",
+        default="False",
+        parser=bool,
+    ),  # Set this to True to always send translations to Smartling
+    "ENVIRONMENT": config(
+        "WAGTAIL_LOCALIZE_SMARTLING_ALWAYS_SEND",
```
Comment (pmac) [inline bedrock/settings/base.py:2135]: This env var is the same as the one above for `REQUIRED`

**Review by pmac — APPROVED** [review]
r+wc

everything other than the question about env vars looks good.

File: docs/cms.rst [hunk 3]
```diff
@@ -112,6 +112,68 @@ This approach will not be a problem if we stick to image filter-specs from the
...
+
+This package supports page-level localization rather than field-level localization, which means that each locale has its own distinct tree of pages, rather than each page having a stack of duplicate fields, one per destination language.
+
+These language-specific trees can be "synchronised" with the default ``en-US`` page tree, so would have the same page structure, field by field) — or they can not be synchronised, so can have their own extra pages, or some specific pages in the tree can be made not "synchronised", while others are.
+
+Basically, there is plenty of flexibility. The flipside of that flexibility is we may also create an edge-case situation that ``wagtail-localize`` won't work with, but we'll have to see and deal with it.
+
+.. note::
+
+    It's worth investing 15 mins in watching the `Wagtail Localize original demo`_ to get a good feel of how it can work.
```
Comment (robhudson) [inline docs/cms.rst:138]: Just noticing the above text looks like it's wrapped whereas this new text isn't.

**Review by robhudson — APPROVED** [review]

**Pull request merged**